sst/include/SimengCoreWrapper.hh

```diff
@@ -0,0 +1,188 @@
+   * everytime a memory request is forwarded by the interface. This functions
+   * acts a callback and invokes SimengMemHandlers on the memory request.
+   */
+  void handleEvent(StandardMem::Request* ev);
+
+  /**
+   * SST supplied MACRO used to register custom SST:Components with
+   * the SST Core.
+   */
+  SST_ELI_REGISTER_COMPONENT(SimengCoreWrapper, "sstsimeng", "simengcore",
+                             SST_ELI_ELEMENT_VERSION(1, 0, 0),
+                             "Simeng core wrapper for SST",
+                             COMPONENT_CATEGORY_PROCESSOR)
+
+  /**
+   * SST supplied MACRO used to document all parameters needed by
+   * a custom SST:Component.
+   */
+  SST_ELI_DOCUMENT_PARAMS(
+      {"simeng_config_path", "Path to SimEng YAML model config file (string)",
+       ""},
+      {"executable_path",
+       "Path to executable binary to be run by SimEng (string)", ""},
+      {"executable_args",
+       "argument to be passed to the executable binary (string)", ""},
+      {"clock", "Clock rate of the SST clock (string)", ""},
+      {"max_addr_memory", "Maximum address that memory can access (int)"}, )
+
+ private:
+  /** Method used to assemble SimEng core. */
+  void fabricateSimengCore();
+
+  /** Method to split the passed executable argument's string into a vector of
+   * individual arguments. */
+  std::vector<std::string> splitArgs(std::string argString);
+
+  // SST properties
+  /**
+   * SST defined output class used to output information to standard output.
+   * This class has in-built method for different levels of severity and can
+   * also be configured to output information like line-number and filename.
+   */
+  SST::Output output_;
+
+  /**
+   * SST clock for the component register with the custom component
+   * during instantiation using the registerClock method provided
+   * by SST.
+   */
+  TimeConverter* clock_;
+
+  /**
+   * SST::Interfaces::StandardMem interface responsible for convering
+   * SST::StandardMem::Request(s) into SST memory events to be passed
+   * down the memory heirarchy.
+   */
+  StandardMem* mem_;
+
+  // Simeng properties
+  /** Reference to the CoreInstance class responsible for creating the core to
+   * be simulated. */
+  std::unique_ptr<simeng::CoreInstance> coreInstance_;
+
+  /** Reference to SimEng core. */
+  std::shared_ptr<simeng::Core> core_;
+
+  /** Path to the YAML configuration file for SimEng. */
+  std::string configPath_;
+
+  /** Path to the executable binary to be simulated by SimEng. */
+  std::string executablePath_;
+
+  /** Arguments to be passed to executable binary. */
+  std::vector<std::string> executableArgs_;
+
+  /** The cache line width for SST. */
+  uint64_t cacheLineWidth_;
+
+  /** Maximum address availbale to SimEng for memory purposes. */
+  uint64_t maxAddrMemory_;
+
+  /** Reference to the SimEng LinuxProcess. */
+  std::shared_ptr<simeng::kernel::LinuxProcess> process_;
```

Comment:
"LinuxProcess_"?

Reply:
Should it be capitalised? wouldn't `linuxProcess_` suffice?

Reply:
Turns out this isn't being used, so removed it.

Reply:
Shouldn't this be resolved?